src/instrumentation_measurement/measurements.py

```diff
@@ -99,19 +102,33 @@ def measure(self, sampleSize, mode="all", evm="geth", nSamples=1):
             header += elem
 
         print(header)
+    elif mode == measure_perf:
+        header = "program_id,sample_id,task_clock,context_switches,page_faults,instructions,branches,branch_misses,L1_dcache_loads,LLC_loads,LLC_load_misses,L1_icache_loads,L1_icache_load_misses,dTLB_loads,dTLB_load_misses,iTLB_loads,iTLB_load_misses"
+        print(header)
+    elif mode == measure_time:
+        header = "program_id,sample_id,real_time_perf,user_time_perf,sys_time_perf,real_time_pure,user_time_pure,sys_time_pure"
+        print(header)
 
 
     for program in self._programs:
       for sample_id in range(nSamples):
         instrumenter_result = None
         if evm == geth:
-          instrumenter_result = self.run_geth(mode, program, sampleSize)
+          if mode == measure_perf:
+            instrumenter_result = self.run_perf_geth(mode, program, sampleSize)
```

Comment:
looks like we can just have `run_geth` and `run_evmone` and branch there (it has the `mode` parameter). You should also be able to use a lot of common code there.

NOTE that `run_evmone` still uses the old `evmc` path, feel free to overwrite with yours, I'm making the same change anyway.